CMake/CheckAVXCompiles.cmake

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CheckCXXSourceCompiles)
+
+set(TEST_FLAG "-mavx512f -mavx512bw -mavx512vl -mavx512vbmi -mavx512vbmi2")
```

Comment:
This only checks if the platform can compile AVX512. It's also important to verify if the CPU Velox is running on has AVX512 support. 3 things to check for these intrinsics
1. platform the lib is compiled on support compiling
2. CPU hardware the lib is running supports it
3. OS supports
We need to check all of them whenever we're using a special instruction set, but for AVX2 and BMI it's almost supported on all x86 CPUs. However, AVX512 is not supported on all processors and therefore the checks. @Yuhta shall we we have a better way to do these checks in a more systematic way?